calling funtions returning void in interpolated strings should fail at compile time

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** V 0.2.2 ecce800
**OS:** Ubuntu 20.04 amd64

**What did you do?**

```
[pancake@hexa v-v]$ v run test.v 
==================
/tmp/v/test.13208342220087598157.tmp.c:10181: error: cannot convert 'void' to 'void *'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

If you were not working with C interop, this is a compiler bug, please report the bug using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

[pancake@hexa v-v]$ cat test.v 
fn hello() {
	println('world')
}

fn main() {
	println('hello ${hello()}')
}
[pancake@hexa v-v]$ 

```

**What did you expect to see?**

V error
 
**What did you see instead?**

GCC error

--- a/vlib/v/checker/check_types.v
+++ b/vlib/v/checker/check_types.v
@@ -437,6 +437,12 @@ pub fn (mut c Checker) string_inter_lit(mut node ast.StringInterLiteral) ast.Typ
 	c.inside_println_arg = true
 	for i, expr in node.exprs {
 		ftyp := c.expr(expr)
+		if ftyp == ast.void_type {
+			c.error('expression does not return a value', expr.position())
+		} else if ftyp == ast.char_type && ftyp.nr_muls() == 0 {
+			c.error('expression returning type `char` cannot be used in string interpolation directly, print its address or cast it to an integer instead',
+				expr.position())
+		}
 		c.fail_if_unreadable(expr, ftyp, 'interpolation object')
 		node.expr_types << ftyp
 		typ := c.table.unalias_num_type(ftyp)

--- a/vlib/v/checker/tests/bad_types_in_string_inter_lit.out
+++ b/vlib/v/checker/tests/bad_types_in_string_inter_lit.out
@@ -1,0 +1,8 @@
+vlib/v/checker/tests/bad_types_in_string_inter_lit.vv:1:12: error: expression does not return a value
+    1 | println('${exit(0)}')
+      |            ~~~~~~~
+    2 | println('${char(48)}')
+vlib/v/checker/tests/bad_types_in_string_inter_lit.vv:2:12: error: expression returning type `char` cannot be used in string interpolation directly, print its address or cast it to an integer instead
+    1 | println('${exit(0)}')
+    2 | println('${char(48)}')
+      |            ~~~~~~~~

--- a/vlib/v/checker/tests/bad_types_in_string_inter_lit.vv
+++ b/vlib/v/checker/tests/bad_types_in_string_inter_lit.vv
@@ -1,0 +1,2 @@
+println('${exit(0)}')
+println('${char(48)}')

--- a/vlib/v/checker/tests/unknown_method_suggest_name.out
+++ b/vlib/v/checker/tests/unknown_method_suggest_name.out
@@ -14,3 +14,9 @@ Did you mean `translate`?
       |            ~~~~~~~~~~~~
    28 |     println('p: $p')
    29 |     println('v: $v')
+vlib/v/checker/tests/unknown_method_suggest_name.vv:30:15: error: expression does not return a value
+   28 |     println('p: $p')
+   29 |     println('v: $v')
+   30 |     println('z: $z')
+      |                  ^
+   31 | }

--- a/vlib/v/checker/tests/unknown_struct_field_suggest_name.out
+++ b/vlib/v/checker/tests/unknown_struct_field_suggest_name.out
@@ -6,3 +6,10 @@ Did you mean `xxxx`?
       |                ~~~~
    12 |         println('x: $x')
    13 |     }
+vlib/v/checker/tests/unknown_struct_field_suggest_name.vv:12:19: error: expression does not return a value
+   10 |     println('p: $p')
+   11 |     for x in p.xxxa {
+   12 |         println('x: $x')
+      |                      ^
+   13 |     }
+   14 | }

